pkg/spanconfig/spanconfig.go

```diff
@@ -44,6 +45,27 @@ type KVAccessor interface {
 	) error
 }
 
+// KVAccessorWithTxn is identical to KVAccessor, but scoped to the given
+// transaction.
+type KVAccessorWithTxn interface {
+	// GetSpanConfigEntriesForWithTxn is the txn scoped version of
+	// KVAccessor.GetSpanConfigEntriesFor.
+	GetSpanConfigEntriesForWithTxn(
+		ctx context.Context,
+		spans []roachpb.Span,
+		txn *kv.Txn,
```

Comment:
Did you consider changing this around so the interface was the same but you curried the transaction in? 

Something like:

```
type TxnKVAccessor interface{
    WithTxn(*kv.Txn) KVAccessor
}
```